homeassistant/components/touchlinesl/__init__.py

```diff
@@ -0,0 +1,55 @@
+"""The Roth Touchline SL integration."""
+
+from __future__ import annotations
+
+from pytouchlinesl import TouchlineSL
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+
+from .const import DOMAIN
+from .coordinator import TouchlineSLModuleCoordinator
+
+PLATFORMS: list[Platform] = [Platform.CLIMATE]
+
+type TouchlineSLConfigEntry = ConfigEntry[dict[str, TouchlineSLModuleCoordinator]]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: TouchlineSLConfigEntry) -> bool:
+    """Set up Roth Touchline SL from a config entry."""
+    account = TouchlineSL(
+        username=entry.data[CONF_USERNAME], password=entry.data[CONF_PASSWORD]
+    )
+
+    coordinators: dict[str, TouchlineSLModuleCoordinator] = {}
```

Comment:
I don't see a significance for the coordinator to be a dict, might as well just be a list